avm/res/resources/deployment-script/main.bicep

```
@@ -0,0 +1,256 @@
+metadata name = 'Deployment Scripts'
+metadata description = 'This module deploys Deployment Scripts.'
+metadata owner = 'Azure/module-maintainers'
+
+// ================ //
+// Definitions      //
+// ================ //
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. Specify the type of lock.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
+}?
+
+type roleAssignmentType = {
+  @description('Required. The name of the role to assign. If it cannot be found you can specify the role definition ID instead.')
+  roleDefinitionIdOrName: string
+
+  @description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
+  principalId: string
+
+  @description('Optional. The principal type of the assigned principal ID.')
+  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device' | null)?
```

```suggestion
  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device')?
```
There are PRs open to fix this also for the rest of the modules + specs. Sorry for that.